apps/cowswap-frontend/src/modules/trade/hooks/useBuildTradeDerivedState.ts

```diff
@@ -13,7 +13,12 @@ import { useTradeUsdAmounts } from 'modules/usdAmount'
 
 import { useSafeMemoObject } from 'common/hooks/useSafeMemo'
 
-export function useBuildTradeDerivedState(stateAtom: Atom<ExtendedTradeRawState>) {
+import { TradeDerivedState } from '../types'
+
+export function useBuildTradeDerivedState(
+  stateAtom: Atom<ExtendedTradeRawState>,
+  supportBuyOrders: boolean,
```

Comment:
As per comment above, a better name could be something like `shouldApplySlippage` or the inverse `usesExactInputAmount` or similar. What do you think?

Reply:
Thanks for the suggestion!
Indeed, I renamed it to `isQuoteBasedOrder` in order to make it more clear.
It's not really about slippage, but about transparent quote amount usage.